modules/solid_mechanics/src/timeintegrators/CentralDifferenceDirect.C

```diff
@@ -0,0 +1,161 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+// MOOSE includes
+#include "Assembly.h"
+#include "CentralDifference.h"
+#include "CentralDifferenceDirect.h"
+#include "MooseTypes.h"
+#include "MooseVariableFieldBase.h"
+#include "NonlinearSystem.h"
+#include "FEProblem.h"
+// libMesh includes
+#include "libmesh/nonlinear_solver.h"
+#include "libmesh/sparse_matrix.h"
+#include "DirichletBCBase.h"
+#include "libmesh/vector_value.h"
+
+registerMooseObject("SolidMechanicsApp", CentralDifferenceDirect);
+
+InputParameters
+CentralDifferenceDirect::validParams()
+{
+  InputParameters params = ExplicitTimeIntegrator::validParams();
+
+  params.addClassDescription(
+      "Implementation of Explicit/Forward Euler without invoking any of the nonlinear solver");
+
+  params.addParam<bool>("use_constant_mass",
+                        false,
+                        "If set to true, will only compute the mass matrix in the first time step, "
+                        "and keep using it throughout the simulation.");
+  params.addParam<TagName>("mass_matrix_tag", "mass", "The tag for the mass matrix");
+
+  MooseEnum solve_type("consistent lumped lump_preconditioned", "lumped");
+  params.setParameters("solve_type", solve_type);
+  params.ignoreParameter<MooseEnum>("solve_type");
+  return params;
+}
+
+CentralDifferenceDirect::CentralDifferenceDirect(const InputParameters & parameters)
+  : ExplicitTimeIntegrator(parameters),
+    _constant_mass(getParam<bool>("use_constant_mass")),
+    _mass_matrix(getParam<TagName>("mass_matrix_tag"))
+{
+  _fe_problem.setUDotRequested(true);
+  _fe_problem.setUDotOldRequested(true);
+  _fe_problem.setUDotDotRequested(true);
+  _fe_problem.setUDotDotOldRequested(true);
+}
+
+void
+CentralDifferenceDirect::computeTimeDerivatives()
+{
+  /*
+  Because this is called in NonLinearSystemBase
+  this should not actually compute the time derivatives.
+  Calculating time derivatives here will cause issues for the
+  solution update.
+  */
+  return;
+}
+
+void
+CentralDifferenceDirect::solve()
+{
+  // Getting the tagID for the mass matrix
+  auto mass_tag = _sys.subproblem().getMatrixTagID(_mass_matrix);
+
+  // Reset iteration counts
+  _n_nonlinear_iterations = 0;
+  _n_linear_iterations = 0;
+
+  _current_time = _fe_problem.time();
+
+  auto & mass_matrix = _nonlinear_implicit_system->get_system_matrix();
+
+  // Compute the mass matrix
+  if (!_constant_mass || (_constant_mass && _t_step == 1))
+    _fe_problem.computeJacobianTag(
+        *_nonlinear_implicit_system->current_local_solution, mass_matrix, mass_tag);
```

Comment:
I see. It looks like you are using the mass matrix tag ID, but you are not using the extra matrix that is added for that tag ID. So right now you will have two matrices in your system, one of which is not used. Maybe you should just add the `MassMatrix` contributions directly into the "system" matrix tag instead of creating the extra matrix with `extra_tag_matrices`

Reply:
Hmm. Perhaps I am confused as to how these systems works. My understanding is that the mass matrix is computed using `computeJacobianTag`. The `MassMatrix` tag is used in `computeJacobianTag` to properly calculate the mass matrix from `MassMatrix.C`. 
 It is then stored in `_nonlinear_implicit_system` as that is where `mass_matrix` is referencing. This leaves open the option to use a constant mass.

Reply:
Does the mass matrix tag ID actually generate a matrix, even without the call to `computeJacobianTag`?

Reply:
Oh I see what you are saying. Ignore the previous comments. That make sense.

Reply:
[9ece25d](https://github.com/idaholab/moose/pull/28175/commits/9ece25dbf1c22cb308158e04798e1903c05d45dc)